tokio-util/src/sync/cancellation_token.rs

```diff
@@ -733,11 +740,13 @@ impl CancellationTokenState {
             // Cancel the child task
             mut_child.cancel();
 
-            // Drop the parent reference. This `CancellationToken` is not interested
-            // in interacting with the child anymore.
-            // This is ONLY allowed once we promised not to touch the state anymore
-            // after this interaction.
-            mut_child.remove_parent_ref(mut_child.snapshot());
+            unsafe {
+                // Drop the parent reference. This `CancellationToken` is not interested
+                // in interacting with the child anymore.
+                // This is ONLY allowed once we promised not to touch the state anymore
+                // after this interaction.
+                Self::remove_parent_ref(mut_child, mut_child.snapshot());
```

Comment:
Here, `mut_child` is also a reference.

Reply:
Whoops, this PR slipped out of my working set.

Per Ralf's example:
```rust
let mut x = Box::new(5);
let xref = &mut *x;
drop(x);
// now xref dangles, but is not used any more. this is fine.
```
do you still want this situation converted to pointers? Happy to make that happen if you want, but if you don't I think it would be best to drop the previous commit.

Reply:
Honestly, my current plan is to just rewrite the entire `CancellationToken`.